### PR TITLE
Fix #7172: Improved App Rating Logic

### LINKED ISSUE #7172: Improved App Rating Logic
https://github.com/brave/internal/issues/1005

### PR DESCRIPTION
**Pull request by soner-yuksel**
## Summary of Changes

This PR is solidifying the variety and changes in App rating Logic and brings a way to support multiple revisions easily, on top adding the new version with its necessary changes.

The review criteria is divided into 2 parts in previous update https://github.com/brave/brave-ios/pull/6210 already and this version is bringing AppReviewLogicType into manager which provides user to assign which criteria must be assigned to which logic type.

It allows us to bring the legacy logic also into this class and combine all logic changes into one place.
The active logic can be easily changed by active logic type in manager.

For new logic please check the ticket for details and Tests can help you navigate for the logic also,

This pull request fixes #7172

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [x] New or updated UI has been tested across:
  - [x] Light & dark mode
  - [x] Different size classes (iPhone, landscape, iPad)
  - [x] Different dynamic type sizes

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: Sources/Growth/AppReviewManager.swift [hunk 1]
```diff
@@ -14,11 +14,61 @@ import BraveVPN
...
+    static let daysInUseRequiredPeriod = 4
+    static let revisedMinimumDaysBetweenReviewRequest = 30
+  }
+  
+  /// A enumeration for which type of App Review Logic will be used
+  /// Helper for quick changes between different types of logic
+  public enum AppReviewLogicType: CaseIterable {
+    case legacy
+    case revised
+    case revisedCrossPlatform
```
Comment (iccub) [inline Sources/Growth/AppReviewManager.swift:52]: Can you add comments for each case explaining them? Hard to tell difference between `revised` and `revisedCrossPlatform`

File: Tests/GrowthTests/AppReviewManagerLegacyTests.swift [hunk 2]
```diff
@@ -0,0 +1,120 @@
+// Copyright 2022 The Brave Authors. All rights reserved.
```
Comment (iccub) [inline Tests/GrowthTests/AppReviewManagerLegacyTests.swift]: 2023

**Review by iccub — CHANGES_REQUESTED** [review]
Add comments to all enum cases for 3 app-review types, rest looks very good to me :)

File: Sources/Growth/AppReviewManager.swift (hunk 1 above)
Comment (soner-yuksel) [reply Sources/Growth/AppReviewManager.swift:52]: Added comments at https://github.com/brave/brave-ios/pull/7201/commits/ee1cd20a0ad4a5bb121231ae3c5a9db53624b614

File: Tests/GrowthTests/AppReviewManagerLegacyTests.swift (hunk 2 above)
Comment (soner-yuksel) [reply Tests/GrowthTests/AppReviewManagerLegacyTests.swift]: Fixed at https://github.com/brave/brave-ios/pull/7201/commits/ee1cd20a0ad4a5bb121231ae3c5a9db53624b614

**Review by iccub — APPROVED** [review]
Thank you

**Pull request merged**